src/ride/track_paint.c

```diff
@@ -982,6 +982,257 @@ TRACK_PAINT_FUNCTION get_track_paint_function_topspin(int trackType, int directi
+	RCT2_GLOBAL(0x141E9B8, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9BC, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C0, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C4, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C8, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9CC, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D0, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D4, uint16) = (height & 0xFF00) | 0x20;
+
+	height += 12;
+	if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) < height) {
+		RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) = height;
+		RCT2_GLOBAL(0x141E9DA, uint8) = 32;
+	}
+}
+
+static void dodgems_paint_setup_rot_1(uint8 rideIndex, uint8 trackSequence, uint8 direction, int height, rct_map_element *mapElement) {
+	dodgems_paint_sub_floor(direction, height);
+
+	switch (trackSequence) {
+		case 0:
+			dodgems_paint_sub_fence_top_right(rideIndex, height, mapElement);
+			dodgems_paint_sub_fence_bottom_right(rideIndex, height, mapElement);
+			break;
+
+		case 1:
+		case 2:
+			dodgems_paint_sub_fence_bottom_right(rideIndex, height, mapElement);
+			break;
+
+		case 3:
+			dodgems_paint_sub_fence_bottom_right(rideIndex, height, mapElement);
+			dodgems_paint_sub_fence_bottom_left(rideIndex, height, mapElement);
+			break;
+
+		case 4:
+		case 8:
+			dodgems_paint_sub_fence_top_right(rideIndex, height, mapElement);
+			break;
+
+		case 7:
+		case 11:
+			dodgems_paint_sub_fence_bottom_left(rideIndex, height, mapElement);
+			break;
+
+		case 12:
+			dodgems_paint_sub_fence_top_left(rideIndex, height, mapElement);
+			dodgems_paint_sub_fence_top_right(rideIndex, height, mapElement);
+			break;
+
+		case 13:
+		case 14:
+			dodgems_paint_sub_fence_top_left(rideIndex, height, mapElement);
+			break;
+
+		case 15:
+			dodgems_paint_sub_fence_top_left(rideIndex, height, mapElement);
+			dodgems_paint_sub_fence_bottom_left(rideIndex, height, mapElement);
+			break;
+	}
+
+	height += 30;
+	if ((trackSequence / 4) & 1) {
+		dodgems_paint_sub_roof(height, 1);
+	} else {
+		dodgems_paint_sub_roof(height, 3);
+	}
+
+	height += 6;
+	RCT2_GLOBAL(0x141E9B4, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9B8, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9BC, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C0, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C4, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9C8, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9CC, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D0, uint16) = (height & 0xFF00) | 0x20;
+	RCT2_GLOBAL(0x141E9D4, uint16) = (height & 0xFF00) | 0x20;
+
+	height += 12;
+	if (RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) < height) {
+		RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_PAINT_TILE_MAX_HEIGHT, sint16) = height;
+		RCT2_GLOBAL(0x141E9DA, uint8) = 32;
+	}
```

Comment:
This if statement is common for a lot of paint functions, so we could turn this into a simple helper.